common/persistence/cassandra/cassandraMetadataPersistence.go

```diff
@@ -97,6 +100,7 @@ type (
 	cassandraMetadataPersistence struct {
 		cassandraStore
 		currentClusterName string
+		thriftEncoder      codec.BinaryEncoder
```

Comment:
Usually we take care of any serialization/deserialization at a layer above and keep the storage layer completely agnostic of encoding/decoding.  Either use strongly typed structures all the way to db or pass in EncodedData.

Reply:
I wanted to do that at the beginning but found that SQL is taking thrift struct and serialize/deserialize inside for the whole workflowExecutionInfo blob. https://github.com/uber/cadence/blob/a19d2cb577aac678663c95482c49b35c739f302c/idl/github.com/uber/cadence/sqlblobs.thrift#L67

But you are also right, maybe we should keep consistent that these small blob should be serialize/deserialize in upper layer. For SQL, I should pass in a blob.  We can talk about that offline.